data-prepper-core/src/test/java/org/opensearch/dataprepper/core/peerforwarder/PeerForwardingProcessingDecoratorTest.java

```diff
@@ -101,6 +106,27 @@ void decorateProcessors_with_different_identification_key_should_throw() {
         assertThrows(RuntimeException.class, () -> createObjectUnderTestDecoratedProcessors(List.of(((Processor) requiresPeerForwarding), (Processor) requiresPeerForwardingCopy)));
     }
 
+    @Test
+    void decorateProcessors_with_excludeIdentificationKeys() {
+        Set<String> identificationKeys = Set.of("key1", "key2");
+        when(requiresPeerForwarding.getIdentificationKeys()).thenReturn(identificationKeys);
+        final List<Processor> processors = createObjectUnderTestDecoratedProcessorsWithExcludeIdentificationKeys(Collections.singletonList((Processor) requiresPeerForwarding), List.of(identificationKeys));
+        for (final Processor processor: processors) {
+            assertTrue(((PeerForwardingProcessorDecorator)processor).isPeerForwardingDisabled());
+        }
+    }
+
+    @Test
+    void decorateProcessors_with_notmatching_excludeIdentificationKeys() {
+        Set<String> identificationKeys = Set.of("key1", "key2");
+        Set<String> notMatchingIdentificationKeys = Set.of("key1", "key3");
+        when(requiresPeerForwarding.getIdentificationKeys()).thenReturn(identificationKeys);
+        final List<Processor> processors = createObjectUnderTestDecoratedProcessorsWithExcludeIdentificationKeys(Collections.singletonList((Processor) requiresPeerForwarding), List.of(notMatchingIdentificationKeys));
+        for (final Processor processor: processors) {
```

Comment:
Please assert the number of processors before this loop. Otherwise, this could result in no verification.

```
assertThat(processors, hasSize(2));
```